cachi2/core/package_managers/pip.py

```diff
@@ -1370,20 +1453,140 @@ def _split_hashes_from_options(cls, options: list[str]) -> tuple[list[str], list
         return hashes, reduced_options
 
 
+def _process_req(
+    req: PipRequirement,
+    requirements_file: PipRequirementsFile,
+    require_hashes: bool,
+    pip_deps_dir: RootedPath,
+    download_info: dict[str, Any],
+    dpi: Optional[DistributionPackageInfo] = None,
+) -> dict[str, Any]:
+    download_info["kind"] = req.kind
+    download_info["requirement_file"] = str(requirements_file.file_path.subpath_from_root)
+    download_info["hash_verified"] = False
```

Comment:
I do not like the fact this function immediately rewrites one of its arguments.

Reply:
I don't understand *exactly* what you mean here - for one thing, the dict arg isn't being rewritten, it's having entries added to it.